File: serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/JwkController.java
package no.difi.meldingsutveksling.serviceregistry.controller;

import com.nimbusds.jose.jwk.JWKSet;
import com.nimbusds.jose.jwk.RSAKey;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class JwkController {

    private final String jwkJson;

    public JwkController(RSAKey rsaKey) {
        this.jwkJson = new JWKSet(rsaKey).toString();
    }

    @GetMapping(value = "/jwk", produces = MediaType.APPLICATION_JSON_VALUE)
    public String jwkEndpoint() {
        return jwkJson;
    }
}

File: serviceregistry-server/src/main/kotlin/no/difi/meldingsutveksling/serviceregistry/controller/JwkController.kt
package no.difi.meldingsutveksling.serviceregistry.controller

import com.nimbusds.jose.jwk.JWKSet
import com.nimbusds.jose.jwk.RSAKey
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class JwkController(rsaKey: RSAKey) {
    private val jwkJson: String = JWKSet(rsaKey).toString(true)

    @GetMapping(value = ["/jwk"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun jwkEndpoint(): String {
        return jwkJson
    }
}